octane-eclipse-plugin/src/com/hpe/octane/ideplugins/eclipse/ui/util/error/ErrorDialog.java

```diff
@@ -63,33 +52,52 @@ private static void positionShell(Shell shell) {
         shell.setLocation(x, y);
     }
 
-    public String open(OctaneException exception, String title) {
-        positionShell(shell);
+    public void displayException(Exception exception, String title) {
+        displayException(exception, title, false);
+    }
+
+    public void displayException(Exception exception, String title, boolean blockThreadUntilClosed) {
         shell.setText(title);
-        exceptionText.setText(exception.getMessage());   
+        errorComposite.displayException(exception);
+        resizeShell();
         shell.open();
-        
-        Display display = getParent().getDisplay();
-        while (!shell.isDisposed()) {
-            if (!display.readAndDispatch()) {
-                display.sleep();
+
+        if(blockThreadUntilClosed) {
+            Display display = getParent().getDisplay();
+            while (!shell.isDisposed()) {
+                if (!display.readAndDispatch()) {
+                    display.sleep();
+                }
             }
         }
-        return null;
     }
-    
+
+    private void resizeShell() {
+        //pack to child size
+        shell.pack();
+
+        //max size 800x600
+        if(shell.getSize().x > MAX_WIDTH) {

```

Comment:
I think something is wrong here because when the error pops up, there is a big white space at the rightmost of the dialog box (even if you resize it, it still persists)